src/libDirectoryService/DirectoryService.cpp

```diff
@@ -507,3 +507,93 @@ std::string DirectoryService::GetActionString(Action action) const
         ? "Unknown"
         : ActionStrings.at(action);
 }
+
+uint8_t
+DirectoryService::CalculateNewDifficulty(const uint8_t& currentDifficulty)
+{
+    const int8_t MAX_ADJUST_STEP = 2;
+    int64_t requiredNodes = 0, powSubmissions = 0;
+    {
+        lock_guard<mutex> g(m_mutexAllPOW);
+        powSubmissions = m_allPoWs.size();
+
+        for (const auto& shard : m_shards)
+        {
+            requiredNodes += shard.size();
+        }
+    }
+
+    LOG_GENERAL(INFO,
+                "requiredNodes " << requiredNodes << ", powSubmissions "
+                                 << powSubmissions);
+
+    int64_t adjustment = 0;
+    if (requiredNodes > 0 && requiredNodes != powSubmissions)
+    {
+        int64_t submissionsDiff;
+        if (!SafeMath<int64_t>::sub(powSubmissions, requiredNodes,
+                                    submissionsDiff))
+        {
+            LOG_GENERAL(WARNING,
+                        "Calculate PoW submission difference goes wrong");
+            submissionsDiff = 0;
+        }
+
+        // To make the adjustment work on small network.
+        auto adjustThreshold
+            = requiredNodes * POW_CHANGE_PERCENT_TO_ADJ_DIFF / 100;
+        if (adjustThreshold > 99)
```

Comment:
Should avoid magic number and suggest to give a meaningful name for "99" or "100"